snap/squashfs2/xz_backend.go

```diff
@@ -0,0 +1,66 @@
+package squashfs2
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/ulikunitz/xz"
```

Comment:
We should probably measure performance of this package, especially if we intend to use it for anything more than reading `snap.yaml` - there are some remarks about performance not being so good (e.g. https://github.com/ulikunitz/xz/issues/30; but of course there is not data to back up that claim, so who knows, maybe it's fine for us).

Reply:
It would probably make sense to have a minimal test for it and run it on our more exotic architectures (s390x, ppc64el) to see if there aren't any unexpected surprises (e.g. with endianess handling?).

Reply:
(I've just noticed the general remark from @alfonsosanchezbeato re performance concerns)